lua/telescope/builtin/files.lua

```diff
@@ -107,6 +107,58 @@ local function prepare_match(entry, kind)
   return entries
 end
 
+files.file_browser = function(opts)
+  opts = opts or {}
+
+  opts.cwd = opts.cwd and vim.fn.expand(opts.cwd) or vim.loop.cwd()
+
+  local files = {}
+  local fd = vim.loop.fs_scandir(opts.cwd)
+  if fd ~= nil then
+    while true do
+      local name, typ = vim.loop.fs_scandir_next(fd)
+      if name == nil then
+        break
+      end
+      if opts.show_hidden or string.sub(name, 1, 1) ~= '.' then
+        if typ == 'directory' then
+          name = name .. '/'
+        end
+
+        table.insert(files, name)
+      end
+    end
+  end
+
+  table.insert(files, 1, '../')
+
+  pickers.new(opts, {
+    prompt_title = 'Find Files',
+    finder = finders.new_table {
+      results = files,
+      entry_maker = make_entry.gen_from_file(opts),
+    },
+    previewer = previewers.cat.new(opts),
+    sorter = conf.file_sorter(opts),
+    attach_mappings = function(prompt_bufnr, map)
+      actions._goto_file_selection:replace(function(_, cmd)
+        local path = actions.get_selected_entry().path
+        actions.close(prompt_bufnr)
+        if string.sub(path, -1) == '/' then
+          require'telescope.builtin'.file_browser(vim.tbl_extend('force', opts, {cwd = path}))
+        else
+          vim.cmd(string.format(':%s %s', cmd, path))
+        end
+      end)
```

Comment:
@tjdevries I think i have a cool new idea for actions. So if i have a dir i wanna open it and if not i just wanna call `_goto_file_selection` as usual. I have two ideas how to do that, interface wise:
```lua
actions._goto_file_selection:enhance ({
  pre = function()
    local path = actions.get_selected_entry().path
    actions.close(prompt_bufnr)
    if string.sub(path, -1) == '/' then
      require'telescope.builtin'.file_browser(vim.tbl_extend('force', opts, {cwd = path}))
      actions.stop_action(prompt_bufnr) -- An action that can always just stop all actions which would run later
    end
    -- If not true we just proceed as usual with pre functions calling _goto_file_selection
  end,
})
```
or
```lua
actions._goto_file_selection:replace_if(function()
  local path = actions.get_selected_entry().path
  return string.sub(path, -1) == '/'
end, function(_, cmd)
  local path = actions.get_selected_entry().path
  actions.close(prompt_bufnr)
  require'telescope.builtin'.file_browser(vim.tbl_extend('force', opts, {cwd = path}))
end)
```
Basically having two function parameters, the first one will return a bool and if that bool is true we will call the second function, if false we will `_goto_file_selection`. So a replace based on a condition.

What do you think?

Reply:
oh, replace if is very cool! I think it's a good idea!